from https://github.com/curl/curl/pull/13251
From 6b88865e264a3d7a85f8a0dafa5e9be5af8ba9f2 Mon Sep 17 00:00:00 2001
From: Daniel McCarney <daniel@binaryparadox.net>
Date: Sun, 31 Mar 2024 16:19:10 -0400
Subject: [PATCH] m4: reposition USE_RUSTLS="yes" for pkg-config

It's necessary to set this var to "yes" _after_ AC_DEFINE and AC_SUBST
in order for a later `test` to pass so that `check_for_ca_bundle=1`
ends up being set. This is in turn required for the default CA
certificate bundle to be set when building w/ rustls & pkg-config.
---
 m4/curl-rustls.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/m4/curl-rustls.m4 b/m4/curl-rustls.m4
index 574496ed41eafe..ef8d7dd4b5505b 100644
--- a/m4/curl-rustls.m4
+++ b/m4/curl-rustls.m4
@@ -137,10 +137,10 @@ if test "x$OPT_RUSTLS" != xno; then
       dnl additional libs may be necessary.  Hope that we
       dnl don't need any.
       LIBS="$SSL_LIBS $LIBS"
-      USE_RUSTLS="yes"
       ssl_msg="rustls"
       AC_DEFINE(USE_RUSTLS, 1, [if rustls is enabled])
       AC_SUBST(USE_RUSTLS, [1])
+      USE_RUSTLS="yes"
       RUSTLS_ENABLED=1
       test rustls != "$DEFAULT_SSL_BACKEND" || VALID_DEFAULT_SSL_BACKEND=yes
     else
